idl.md

```diff
@@ -1,61 +1,90 @@
-  readonly attribute unsigned long minAllowablePeriod;
+  Promise<SyncPermissionState> permissionState();
 };
 
 interface SyncRegistration {
-  readonly attribute DOMString id;
-  readonly attribute unsigned long minDelay;
-  readonly attribute unsigned long maxDelay;
-  readonly attribute unsigned long minPeriod;
-  readonly attribute SyncNetworkType minRequiredNetwork;
-  readonly attribute boolean allowOnBattery;
-  readonly attribute boolean idleRequired;
+  readonly attribute DOMString tag;
 
-  Promise <boolean> unregister();
+  Promise<boolean> unregister();
 };
 
 dictionary SyncRegistrationOptions {
-  DOMString id = "";
-  unsigned long minDelay = 0;
-  unsigned long maxDelay = 0;
-  unsigned long minPeriod = 0;
-  SyncNetworkType minRequiredNetwork = "network-online";
-  boolean allowOnBattery = true;
-  boolean idleRequired = false;
-};
-
-enum SyncNetworkType {
-  "network-offline",
-  "network-online",
-  "network-non-mobile",
+  DOMString tag = "";
 };
 
-enum SyncPermissionStatus {
+enum SyncPermissionState {
   "default",
   "denied",
   "granted"
 };
 
-partial interface ServiceWorkerGlobalScope {
-  attribute EventHandler onsync;
+enum SyncNetworkState {
+  "any",
+  "avoid-cellular",
+  "online"
 };
 
-dictionary SyncEventInit : EventInit {
-  required SyncRegistration registration; // this is mostly a no-op for now
+enum SyncPowerState {
+  "auto",
+  "avoid-draining"
+};
+
+interface PeriodicSyncManager {
+  Promise<PeriodicSyncRegistration> register(optional PeriodicSyncRegistrationOptions options);
+  Promise<PeriodicSyncRegistration> getRegistration(DOMString tag);
+  Promise<sequence<PeriodicSyncRegistration>> getRegistrations();
+  Promise<SyncPermissionState> permissionState();
+
+  readonly attribute unsigned long minPossiblePeriod;
+};
+
+interface PeriodicSyncRegistration {
+  readonly attribute DOMString tag;
+  readonly attribute unsigned long minPeriod;
+  readonly attribute SyncNetworkState networkState;
+  readonly attribute SyncPowerState powerState;
+
+  Promise<boolean> unregister();
+};
+
+dictionary PeriodicSyncRegistrationOptions {
+  DOMString tag = "";
+  unsigned long minPeriod = 0;
+  SyncNetworkType networkState = "online";
+  SyncPowerState powerState = "auto";
+};
+
+partial interface ServiceWorkerGlobalScope {
+  attribute EventHandler onsync;
+  attribute EventHandler onperiodicsync;
 };
 
 [Constructor(DOMString type, SyncEventInit eventInitDict), Exposed=ServiceWorker]
 interface SyncEvent : ExtendableEvent {
   readonly attribute SyncRegistration registration;
 };
+
+dictionary SyncEventInit : EventInit {
+  required SyncRegistration registration;
+};
+
+[Constructor(DOMString type, PeriodicSyncEventInit eventInitDict), Exposed=ServiceWorker]
+interface PeriodicSyncEvent : ExtendableEvent {
+  readonly attribute PeriodicSyncRegistration registration;
+};
+
+dictionary PeriodicSyncEventInit : EventInit {
+  required SyncRegistration registration;
```

Comment:
That should say `PeriodicSyncRegistration` I think. However, perhaps we could have that inherit from `SyncRegistration` and make do with only a single event class (and have the distinction only in the type of the event).